### ISSUE TITLE
T  - BUG: Create doc throws an error

### ISSUE THREAD

**Issue by tobinmori** [mdn/yari#794]
When I try to create doc using the toolbar I get:

YAMLException: unacceptable kind of an object to dump [object Undefined]

**Comment by tobinmori** [comment]
Error saving document

Error: 500: <!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>YAMLException: unacceptable kind of an object to dump [object Undefined]<br> &nbsp; &nbsp;at writeNode (/Users/tmori/code/yari/yari-fork/yari/node_modules/js-yaml/lib/js-yaml/dumper.js:779:13)<br> &nbsp; &nbsp;at writeBlockMapping (/Users/tmori/code/yari/yari-fork/yari/node_modules/js-yaml/lib/js-yaml/dumper.js:657:10)<br> &nbsp; &nbsp;at writeNode (/Users/tmori/code/yari/yari-fork/yari/node_modules/js-yaml/lib/js-yaml/dumper.js:750:9)<br> &nbsp; &nbsp;at dump (/Users/tmori/code/yari/yari-fork/yari/node_modules/js-yaml/lib/js-yaml/dumper.js:840:7)<br> &nbsp; &nbsp;at Object.safeDump (/Users/tmori/code/yari/yari-fork/yari/node_modules/js-yaml/lib/js-yaml/dumper.js:846:10)<br> &nbsp; &nbsp;at saveHTMLFile (/Users/tmori/code/yari/yari-fork/yari/content/scripts/document.js:36:33)<br> &nbsp; &nbsp;at Object.create (/Users/tmori/code/yari/yari-fork/yari/content/scripts/document.js:63:3)<br> &nbsp; &nbsp;at /Users/tmori/code/yari/yari-fork/yari/server/document.js:30:12<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/Users/tmori/code/yari/yari-fork/yari/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at next (/Users/tmori/code/yari/yari-fork/yari/node_modules/express/lib/router/route.js:137:13)</pre>
</body>
</html>

**Comment by tobinmori** [comment]
YAMLException: unacceptable kind of an object to dump [object Undefined]

**Comment by peterbe** [comment]
I fixed it as part of https://github.com/mdn/yari/pull/730
Yeah, it was a bit ugly to lump that code fix together with all the content changes. But here we are. And I didn't want to get blocked. 

The solution I did can be seen here: https://github.com/mdn/yari/blame/master/content/scripts/document.js
Basically, instead of doing:
```javascript
yaml.safeDump({title, summary, slug, tags})
```
it now does:
```javascript
const metadata = {title, summary, slug}
if (tags) {
  metadata.tags = tags
}
yaml.safeDump(metadata)
```

@Gregoor If you think that's sane enough feel free to close this issue.
